shell/utils/fleet.ts

```diff
@@ -0,0 +1,106 @@
+import { BundleDeploymentResource, ModifiedStatus, NonReadyBundle } from '@shell/types/resources/fleet';
+import { STATES_ENUM } from '@shell/plugins/dashboard-store/resource-class';
+import { FLEET as FLEET_ANNOTATIONS } from '@shell/config/labels-annotations';
+
+type state = STATES_ENUM.READY | STATES_ENUM.MODIFIED | STATES_ENUM.MISSING | STATES_ENUM.ORPHANED;
+
+interface Resource extends BundleDeploymentResource {
+  state: state,
+}
+
+interface BundleDeploymentStatus {
+  resources?: BundleDeploymentResource[],
+  modifiedStatus?: ModifiedStatus[],
+}
+
+interface BundleStatusSummary {
+  nonReadyResources?: NonReadyBundle[],
+}
+
+interface BundleStatus {
+  resourceKey?: BundleDeploymentResource[],
+  summary?: BundleStatusSummary,
+}
+
+// bundleDeploymentResources extracts the list of resources deployed by a BundleDeployment
+export function bundleDeploymentResources(status: BundleDeploymentStatus): Resource[] {
+  // Use a map to avoid `find` over and over again
+  const resourceKey = (r) => `${ r.kind }/${ r.namespace }/${ r.name }`;
```

Comment:
`r: any` to avoid TS error, though might be worth creating 

```
interface StatusResource {
  apiVersion: string,
  kind: string,
  name: string,
}
```